crypto/stack/stack.c

```diff
@@ -322,17 +327,39 @@ static int internal_find(OPENSSL_STACK *st, const void *data,
     r = ossl_bsearch(&data, st->data, st->num, sizeof(void *), st->comp,
                      ret_val_options);
 
+    if (pnum != NULL) {
+        if (r != NULL) {
```

Comment:
Consider hoisting the `*pnum = 0` up to here and removing the else clause?

Reply:
OK